packages/repository/src/mixins/repository.mixin.ts

```diff
@@ -200,11 +202,16 @@ export function RepositoryMixin<T extends MixinTarget<Application>>(
      *
      * @param component - The component to mount repositories of
      */
-    mountComponentRepositories(component: Class<unknown>) {
-      const componentKey = `${CoreBindings.COMPONENTS}.${component.name}`;
-      const compInstance = this.getSync<{
-        repositories?: Class<Repository<Model>>[];
-      }>(componentKey);
+    mountComponentRepositories(
+      component: Class<unknown> | {repositories?: Class<Repository<Model>>[]},
+    ) {
+      // accept also component class to preserve backwards compatibility
+      const compInstance =
+        typeof component === 'function'
+          ? this.getComponentInstance<{
+              repositories?: Class<Repository<Model>>[];
+            }>(component)
+          : component;
```

Comment:
This is not needed if https://github.com/strongloop/loopback-next/pull/5477/commits/1cf38b14016dd60bb3ff43b3b2f1ede7dd4e5972#r426750582 is addressed.

Reply:
I am afraid the link does not point to any comment :(

I believe this is still needed to support the following use case:

```ts
app.mountComponentRepositories(SomeComponentClass);
```

(Note that `mountComponentRepositories` is a public method.)